cabal-install/Distribution/Client/List.hs

```diff
@@ -129,6 +138,94 @@ getPkgList verbosity packageDBs repoCtxt comp conf listFlags pats = do
       , pkg <- search index pat ]
 
 
+-- | List the installed packages that the current package depends on.
+-- This is similar to getPkgList, but it takes no patterns; rather, it assumes
+-- there is a current, configured configuration so it can determine the exact
+-- (installed) packages depended on.
+listPkgDeps :: Verbosity
+            -> RepoContext
+            -> ListFlags
+            -> IO ()
+listPkgDeps verbosity repoCtxt listFlags =
+    case possiblyShowPackageFields of
+      Left msg                -> die msg
+      Right showPackageFields -> do
+          installedPkgIndex <-
+            fmap Distribution.Simple.LocalBuildInfo.installedPkgs $
+              getPersistBuildConfig (useDistPref defaultSetupScriptOptions)
+               -- TODO: add a --builddir=DIR option rather than assume
+               -- the default value of 'dist'
+
+          srcPkgIndex <- fmap packageIndex $
+                           getSourcePackages verbosity repoCtxt
+
+          let
+              showPkgSimple pkg =
+                display (packageName pkg)
+                ++ " "
+                ++ display (packageVersion pkg)
+                ++ showPackageFields pkg
+
+              showPkgStd = showPackageSummaryInfo . mkDisplayInfo srcPkgIndex
+
+              pkgs = InstalledPackageIndex.topologicalOrder installedPkgIndex
```

Comment:
I don't think that this is correct, it will print all packages in the package index. I think you should use `dependencyClosure`.